File: examples/zig/test_mle.zig
const std = @import("std");

pub fn main() !void {
    const allocator = std.heap.page_allocator;
    
    while (true) {
        // Allocate 10MB chunks
        const mem = try allocator.alloc(u8, 10 * 1024 * 1024);
        // Write to the memory to ensure the OS maps physical RAM
        @memset(mem, 1);
    }
}

File: examples/c/test_mle.c
#include <stdlib.h>
#include <string.h>

int main() {
    while (1) {
        // Allocate 10MB at a time
        void *p = malloc(1024 * 1024 * 10);
        if (p == NULL) break;
        memset(p, 0, 1024 * 1024 * 10); // Force physical allocation
    }
    return 0;
}